pytorch_pfn_extras/training/_evaluator.py

```diff
@@ -151,8 +155,29 @@ def run(
                         prof.step()  # type: ignore[no-untyped-call]
         # This will report to the trainer main reporter
         self.handler.eval_loop_end(self)
+        self._gather_summaries()
         reporting.report(self._summary.compute_mean())
 
 
 # For backward compatibility
 _Evaluator = Evaluator
+
+
+class DistributedEvaluator(Evaluator):
+    def __init__(
+            self,
+            handler: 'BaseHandler',
+            models: Union[torch.nn.Module, Mapping[str, torch.nn.Module]],
+            *,
+            progress_bar: bool = False,
+            metrics: Optional[Sequence['MetricType']] = None,
+    ):
+        super().__init__(handler, models, progress_bar=progress_bar, metrics=metrics)
+        if not torch.distributed.is_initialized():  # type: ignore[no-untyped-call]
+            raise RuntimeError("PyTorch distributed module is not initialized.")
+
+    def _gather_summaries(self) -> None:
+        world_size = torch.distributed.get_world_size()  # type: ignore[no-untyped-call]
+        summaries = [reporting.DictSummary() for _ in range(world_size)]
+        torch.distributed.all_gather_object(summaries, self._summary)  # type: ignore[no-untyped-call]
```

Comment:
I am wondering if instead of getting the summaries, what would be the disadvantages of specifying a custom `key` that is in all summaries and perform an AllReduce op based in those?

```python
eval = DistributedEvaluator(gather_keys=["loss", "accuracy"])

class DistributedEvaluator(Evaluator):
    def _gather_summaries(self) -> None:
         for k in self._gather_keys:
             self._summary[k] = torch.distributed.all_reduce(self._summary[k], ...)
```
The problem I see with this is that we may need several reduction functions, but seems that summary only does means of values so it should be generic enough and we can implement it without modifying summary?

This is just a non-well thought idea so feel free to ignore it if it doesn't make sense :)